### PR TITLE
@visitha 's send email feature

### PR DESCRIPTION
**Pull request by georgehrke**

### CODE REVIEW

**Comment by georgehrke** [comment]
@Raydiation @Kondou-ger
you reviewed this earlier. Is there anything left you would change?

File: lib/app.php [hunk 1]
```diff
@@ -511,4 +511,54 @@ public static function generateEventOutput(array $event, $start, $end) {
...
+               $subject = 'Calendar Event Shared';
+               
+               $headers = 'MIME-Version: 1.0\r\n';
+               $headers .= 'Content-Type: text/html; charset=utf-8\r\n';
+               $headers .= 'From:' . $useremail;
+
+               $message  = '<html><body>';
+               $message .= '<table style="border:1px solid black;" cellpadding="10">';
+               $message .= "<tr style='background: #eee;'><td colspan='2'><strong>" . $user . '</strong><strong> has shared with you an event</strong></td></tr>';
+               $message .= '<tr><td><strong>Location:</strong> </td><td>' . $location . '</td></tr>';
```
Comment (BernhardPosselt) [inline lib/app.php]: XSS

File: lib/app.php [hunk 2]
```diff
@@ -511,4 +511,54 @@ public static function generateEventOutput(array $event, $start, $end) {
...
+               $message .= '<tr><td><strong>Description:</strong> </td><td>' . $description . '</td></tr>';
```
Comment (BernhardPosselt) [inline lib/app.php]: XSS

File: lib/app.php (hunk 1 above)
Comment (georgehrke) [inline lib/app.php]: p() won't work in this case.
What's the alternative?

Comment (Niduroki) [inline lib/app.php]: `\OC_Util::sanitizeHTML()` (public interface version of that)

Comment (BernhardPosselt) [inline lib/app.php]: https://www.lazycat.org/php-output-escaping.html

**Comment by BernhardPosselt** [comment]
:+1:

**Comment by georgehrke** [comment]
@Raydiation @Kondou-ger I fixed some more issues. Do you have time to give it a final try? :)

**Comment by Niduroki** [comment]
```
An exception occurred while executing 'SELECT "id", "item_type", "item_source", "parent", "share_type", "share_with", "permissions", "stime", "file_source", "expiration", "token", "mail_send" FROM "oc_share" WHERE "item_type" = ? AND "uid_owner" = ? AND "share_type" != ? AND "item_source" = ?':SQLSTATE[42703]: Undefined column: 7 ERROR: column "mail_send" does not existLINE 1: ...", "stime", "file_source", "expiration", "token", "mail_send... ^
#0 /home/http/testcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Statement.php(140): Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Object(PDOException), 'SELECT "id", "i...', Array)#1
```

**Comment by georgehrke** [comment]
@Kondou-ger Where is the failing sql command? I can't find any sql command in the "files changed" tab

**Comment by Niduroki** [comment]
@georgehrke nvm, cookies didn't update, so I couldn't update my owncloud instance …

There's no non-html email text, in the sent email.

File: lib/app.php [hunk 3]
```diff
@@ -511,4 +511,58 @@ public static function generateEventOutput(array $event, $start, $end) {
...
+
+			$subject = 'Calendar Event Shared';
+
+			$headers = 'MIME-Version: 1.0\r\n';
+			$headers .= 'Content-Type: text/html; charset=utf-8\r\n';
+			$headers .= 'From:' . $useremail;
+
+			$message  = '<html><body>';
+			$message .= '<table style="border:1px solid black;" cellpadding="10">';
+			$message .= "<tr style='background: #eee;'><td colspan='2'><strong>" . $user . '</strong><strong> has shared with you an event</strong></td></tr>';
```
Comment (Niduroki) [inline lib/app.php:558]: Rename to "shared an event with you"?

**Comment by Niduroki** [comment]
Otherwise: :+1:

**Comment by georgehrke** [comment]
ok, these are things we can fix later. Today is feature freeze, so let's get this merged

**Comment by undigo** [comment]
I try to give the start date and end date to people that I send email with share event calendar function,
so on /lib/apps.php I added these lines in public static function sendEmails($eventid, $summary, $location, $description, $dtstart, $dtend) to append the complete message vy sendemail function :

$message .= '<tr><td><strong>Date et heure de début :</strong> </td><td>' . \OCP\Util::sanitizeHTML($$dtstart) . '</td></tr>';
$message .= '<tr><td><strong>Date et heure de fin :</strong> </td><td>' . \OCP\Util::sanitizeHTML($$dtend) . '</td></tr>';

but when I receive my email these fields are twice empty !!!

I don't know the full workflow for send email with the calendar module.

If you know why these fields are empty ?

Thanks.

**Pull request merged**